src/gui/layertree/qgslayertreeview.h

```diff
@@ -234,6 +234,8 @@ class GUI_EXPORT QgsLayerTreeView : public QTreeView
     void onCurrentChanged();
     void onExpandedChanged( QgsLayerTreeNode *node, bool expanded );
     void onModelReset();
+    //! Handles updating the viewport to avoid flicker
+    void onHorizontalScroll( int value );
```

Comment:
I think this should be in the private slots

Reply:
Fixed in https://github.com/qgis/QGIS/pull/35461/commits/6d1b8d6d063084e12168778f4b50e1df4c685c9d